src/Consolonia.PlatformSupport/Win32Console.cs

```diff
@@ -111,37 +134,123 @@ public override void PauseIO(Task task)
 
         private void StartEventLoop()
         {
-            Task.Run(() =>
+            Task.Run(async () =>
             {
                 while (!Disposed /*inject ThreadAbortException*/)
                 {
                     PauseTask?.Wait();
-                    var readConsoleInput = _windowsConsole.ReadConsoleInput();
-                    if (!readConsoleInput.Any())
-                        throw new NotImplementedException();
-                    foreach (INPUT_RECORD inputRecord in readConsoleInput)
-                        // ReSharper disable once SwitchStatementMissingSomeEnumCasesNoDefault
-                        switch (inputRecord.EventType)
+                    var inputRecords = _windowsConsole.ReadConsoleInput();
+                    var clipboard = AvaloniaLocator.Current.GetService<IClipboard>();
+                    if (clipboard != null &&
+                        inputRecords.Where(evt => evt.EventType == EVENT_TYPE.KEY_EVENT).Skip(1).Any())
+                        // when console is translating CTRL+V to sequence of key strokes it comes in as multiple key events.
+                        await ProcessClipboardInput(clipboard, inputRecords);
+                    else
+                        foreach (INPUT_RECORD inputRecord in inputRecords)
+                            HandleInputRecord(inputRecord);
+                }
+            });
+        }
+
+        /// <summary>
+        ///     Process clipboard input and compare to clipboard text to determine if we should paste clipboard text.
+        /// </summary>
+        /// <param name="clipboard"></param>
+        /// <param name="inputRecords"></param>
+        /// <returns></returns>
+        private async Task ProcessClipboardInput(IClipboard clipboard, INPUT_RECORD[] inputRecords)
+        {
+            string clipboardText = await clipboard?.GetTextAsync() ?? string.Empty;
+            if (clipboardText.Trim().Length == 0)
+            {
+                // no text in clipboard, just process input records
+                foreach (INPUT_RECORD inputRecord in inputRecords)
+                    HandleInputRecord(inputRecord);
```

Comment:
It's code duplication. If we change this logic in the future, which is very likely, we would need to maintain this code twice. I'm proposing to return false in this case and let the caller execute else statement